example3/readme.md

```diff
@@ -659,6 +659,9 @@ Given pass experiences, handling and management of LabelSet will be a hot topic
         );
     ```
 
+## How should we manage the lifetime of instruments/meters?
+
+We probably want to clean them up as some point so we should figure out when that happens. Does Meter or Instrument implement IDisposable?

```

Comment:
I think we need to have IDisposable.
I think the instrument lifecycle is attached to meter - if meter gets disposed, all the instruments "under" that meter should also be disposed. Instrument can also be disposed individually.

Reply:
What if we said Meter is the only thing that is IDisposable and if you want counters to have different lifetimes they should be in a different Meter, does that feel too restrictive?

Reply:
I think it is perfectly fine.
Curious - is there a reason you would prefer not to have IDisposable on Instruments (Counter, Gauge, ...)?

Reply:
The usual expectation is that any type implementing IDisposable needs to be disposed. I was hoping to avoid making users think they need to call Dispose() on every instrument individually when disposing the Meter alone would work just as well.